### ISSUE TITLE
Language Clarification:  Concurrently accessing the physics simulation

### ISSUE THREAD

**Issue by moonshineTheleocat** [jrouwe/JoltPhysics#95]
> In games we usually need to do many more things than to simulate the physics world and we need to do this across multiple threads. We therefore place a lot of emphasis on concurrently accessing the physics simulation data outside of the main physics simulation update:

Just asking for a clarification here, as the language makes it a bit confusing. Is Jolt intended to be run on it's own thread, while sharing a thread pool with the main application thread. Or is it designed to run on the main application thread.

**Comment by jrouwe** [comment]
The simulation runs across many threads (doesn't matter if you kick it from the main thread or another thread, it will use that thread and as many other threads as you give it - this is abstracted in the JobSystem class with JobSystemThreadPool as a possible implementation). The text above is talking about what happens outside of the physics simulation: You can manipulate bodies and perform collision queries from many threads at the same time.

**Comment by moonshineTheleocat** [comment]
Ok. So the main physics loop can exist on the application's main thread, or continuously running on its own thread, and it can be handled just fine provided it has interface access to the application's thread-pool implementation.

**Comment by jrouwe** [comment]
Correct. I'm assuming b.t.w. that 'continuously running' means that the thread continuously runs, not the simulation. During PhysicsSystem::Update the physics system is locked, so other threads cannot read/modify any state. Outside of the PhysicsSystem::Update multiple threads can access and modify the simulation data if you want.

**Comment by moonshineTheleocat** [comment]
Gotcha, gotcha. That basically answers the question. So I'm gonna go ahead and lock the thread.
